src/objects/Button.ts

```diff
@@ -0,0 +1,65 @@
+import * as Phaser from 'phaser';
+
+type ButtonOptionsType = {
+  content: string;
+  width: number;
+  onClick: () => void;
+};
+
+class Button extends Phaser.GameObjects.Container {
+  public button: Phaser.GameObjects.NineSlice;
+
+  static preload(scene: Phaser.Scene) {
+    scene.load.atlas(
+      'ui',
+      'https://labs.phaser.io/assets/ui/nine-slice.png',
+      'https://labs.phaser.io/assets/ui/nine-slice.json',
+    );
+  }
+
+  constructor(
+    scene: Phaser.Scene,
+    x: number,
+    y: number,
+    options: ButtonOptionsType,
+  ) {
+    super(scene, x, y);
+
+    this.scene = scene;
+
+    const { content, width, onClick } = options;
+
+    this.button = scene.add.nineslice(
+      x,
+      y,
+      'ui',
+      'button-bg',
+      128,
+      110,
+      64,
+      64,
+    );
+
+    const text = scene.add.text(x, y, content, {
+      font: '25px Arial',
+      align: 'center',
+    });
+    text.setOrigin(0.5, 0.5);
+    text.setWordWrapWidth(width - 100);
+
+    this.button.width = width;
+    this.button.setInteractive({ useHandCursor: true });
+    this.button.on('pointerdown', onClick);
+
+    scene.tweens.add({
+      targets: [this.button],
```

Comment:
Nice bit of tweening! 

not a big issue but I noticed the collision box doesn't scale with the tweening when increasing size.  Do we also need to target the hitbox as well?

Reply:
just lifted that from the example pretty much verbatim
https://phaser.io/examples/v3/view/game-objects/nine-slice/3-slice-button

i think ideally the hitarea of this button should be much bigger, like almost cover the entire screen if possible, but not trigger when clicking the settings cog icon or it's children

Reply:
ive fixed the hitArea placement for now, but not added it to the tween, idk how to :shrug: 
i tried a few things

Reply:
Sweet, yeah only a minor thing we can look at another time